h5io/_h5io.py

```diff
@@ -211,6 +231,8 @@ def _triage_read(node):
         if type_str == 'dict':
             data = dict()
             for key, subnode in node.items():
+                for key_spec, val_spec in special_chars.items():
```

Comment:
We should only do this if `slash='replace'` in `read_hdf5` (and you'll need to add that as an option). The default there should be `'ignore'` just in case someone might have `{FWDSLASH}` in their name for some other reason.